lib/spack/spack/spec.py

```diff
@@ -1868,6 +1868,7 @@ def _evaluate_dependency_conditions(self, name):
         pkg = spack.repo.get(self.fullname)
         conditions = pkg.dependencies[name]
 
+        substitute_abstract_variants(self)
```

Comment:
I think this is the right thing to do. When implementing MV variants I was giving for granted that here all the abstract variants in the package were substituted already.